silverback/middlewares.py

```diff
@@ -66,34 +64,40 @@ def fix_dict(data: dict, recurse_count: int = 0) -> dict:
         return message
 
     def _create_label(self, message: TaskiqMessage) -> str:
-        if message.task_name == "block":
-            args = f"[block={message.args[0].hash.hex()}]"
-
-        elif "event" in message.task_name:
-            args = f"[txn={message.args[0].transaction_hash},log_index={message.args[0].log_index}]"
+        if labels_str := (
+            ",".join(f"{k}={v}" for k, v in message.labels.items() if k != "task_name")
+        ):
+            return f"{message.task_name}[{labels_str}]"
 
         else:
-            args = ""
-
-        return f"{message.task_name}{args}"
+            return message.task_name
 
     def pre_execute(self, message: TaskiqMessage) -> TaskiqMessage:
-        if message.task_name == "block":
+        message.labels["task_name"] = message.task_name
+        task_type = message.labels.pop("task_type", "<unknown>")
+
+        # NOTE: Don't compare `str` to `TaskType` using `is`
+        if task_type == TaskType.NEW_BLOCKS:
             # NOTE: Necessary because we don't know the exact block class
             message.args[0] = self.provider.network.ecosystem.decode_block(
                 hexbytes_dict(message.args[0])
             )
+            message.labels["number"] = str(message.args[0].number)
+            message.labels["hash"] = message.args[0].hash.hex()
 
         elif "event" in message.task_name:
             # NOTE: Just in case the user doesn't specify type as `ContractLog`
             message.args[0] = ContractLog.model_validate(message.args[0])
+            message.labels["block"] = str(message.args[0].block_number)
+            message.labels["txn_id"] = message.args[0].transaction_hash
```

Comment:
suggestion: maybe keep the terms the same, rather than abbreviating?

Reply:
Thought about that, could go either way, but it really does blow up the size of the logs that get generated. Probably the consistency is more worth it though

Reply:
I actually came back and realized I could remove a bunch more stuff that isn't really needed anymore for persistence